stubs/tqdm/tqdm/std.pyi

```diff
@@ -0,0 +1,272 @@
+        ncols: int | None = ...,
+        prefix: str | None = ...,
+        ascii: bool | str | None = ...,
+        unit: str | None = ...,
+        unit_scale: bool | float | None = ...,
+        rate: float | None = ...,
+        bar_format: str | None = ...,
+        postfix: str | Mapping[object, object] | None = ...,
+        unit_divisor: float | None = ...,
+        initial: float | None = ...,
+        colour: str | None = ...,
+    ) -> str: ...
+    @overload
+    def __init__(
+        self,
+        iterable: Iterable[_T] | None,
+        desc: str | None = ...,
+        total: float | None = ...,
+        leave: bool = ...,
+        file: str | io.TextIOWrapper | io.StringIO | None = ...,
+        ncols: int | None = ...,
+        mininterval: float = ...,
+        maxinterval: float = ...,
+        miniters: float | None = ...,
+        ascii: bool | str | None = ...,
+        disable: bool = ...,
+        unit: str = ...,
+        unit_scale: bool | float = ...,
+        dynamic_ncols: bool = ...,
+        smoothing: float = ...,
+        bar_format: str | None = ...,
+        initial: float = ...,
+        position: int | None = ...,
+        postfix: Mapping[object, object] | str | None = ...,
+        unit_divisor: float = ...,
+        write_bytes: bool | None = ...,
+        lock_args: tuple[bool | None, float | None] | tuple[bool | None] | None = ...,
+        nrows: int | None = ...,
+        colour: str | None = ...,
+        delay: float | None = ...,
+        gui: bool = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        desc: str | None = ...,
+        total: float | None = ...,
+        leave: bool = ...,
+        file: str | io.TextIOWrapper | io.StringIO | None = ...,
+        ncols: int | None = ...,
+        mininterval: float = ...,
+        maxinterval: float = ...,
+        miniters: float | None = ...,
+        ascii: bool | str | None = ...,
+        disable: bool = ...,
+        unit: str = ...,
+        unit_scale: bool | float = ...,
+        dynamic_ncols: bool = ...,
+        smoothing: float = ...,
+        bar_format: str | None = ...,
+        initial: float = ...,
+        position: int | None = ...,
+        postfix: Mapping[object, object] | str | None = ...,
+        unit_divisor: float = ...,
+        write_bytes: bool | None = ...,
+        lock_args: tuple[bool | None, float | None] | tuple[bool | None] | None = ...,
+        nrows: int | None = ...,
+        colour: str | None = ...,
+        delay: float | None = ...,
+        gui: bool = ...,
+    ) -> None: ...
+    def __new__(cls, *_, **__): ...
+    @classmethod
+    def write(
+        cls, s: str, file: str | io.TextIOWrapper | io.StringIO | None = ..., end: str = ..., nolock: bool = ...
+    ) -> None: ...
+    @classmethod
+    def external_write_mode(
+        cls, file: str | io.TextIOWrapper | io.StringIO | None = ..., nolock: bool = ...
+    ) -> AbstractContextManager[None]: ...
```

Comment:
`file` here should probably also be `SupportsWrite[str]`. Also, we prefer precise return types in typeshed, so it would be better to return `contextlib._GeneratorContextManager[None]` rather than `AbstractContextManager[None]`.